src/app/services/sortFunctions/sort.service.ts

```diff
@@ -0,0 +1,47 @@
+import {Injectable} from '@angular/core';
+import {ISelected} from "./iselected";
+import {Movie} from "../../model/movie/movie";
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SortService {
+  private _sortFields : ISelected[] = [
+    {
+      title: "Votes",
+      value: (a: Movie, b: Movie): number => a.vote_average - b.vote_average
+    },
+    {
+      title: "Release date",
+      value: (a: Movie, b: Movie): number => {
+        if (a.release_date > b.release_date) {
+          return 1;
+        } else if (a.release_date < b.release_date) {
+          return -1;
+        } else
+          return 0;
+      }
+    },
+    {
+      title: "Budget",
+      value: (a: Movie, b: Movie): number => a.budget - b.budget
+    },
+    {
+      title: "Revenue",
+      value: (a: Movie, b: Movie): number => a.revenue - b.revenue
+    },
+    {
+      title: "Runtime",
+      value: (a: Movie, b: Movie): number => a.runtime - b.runtime
```

Comment:
you can make it a higher order function which will accept by which field to sort, so to practice with it and it'll remove these "value" functions repetitions